File: ExampleA0/app/src/main/java/com/abubusoft/kripton/examplea0/model/DateAdapter.java
package com.abubusoft.kripton.examplea0.model;

import com.abubusoft.kripton.BindTypeAdapter;

import java.util.Date;

/**
 * Created by xcesco on 29/08/2017.
 */

public class DateAdapter implements BindTypeAdapter<Date, Long> {

    @Override
    public Date toJava(Long dataValue) throws Exception {
        if (dataValue==null) return null;
        return new Date(dataValue);
    }

    @Override
    public Long toData(Date javaValue) throws Exception {
        if (javaValue==null) return null;
        return javaValue.getTime();
    }
}

File: kotlin-rss-reader/app/src/main/java/com/abubusoft/kripton/example/rssreader/service/model/DateAdapter.kt
package com.abubusoft.kripton.example.rssreader.service.model

import com.abubusoft.kripton.TypeAdapter
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

class DateAdapter : TypeAdapter<Date, String> {

    private var formatter: SimpleDateFormat = SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss zzz", Locale.ENGLISH);

    override fun toJava(dataValue: String?): Date? {
        if (dataValue == null) return null
        return try {
            formatter.parse(dataValue)
        } catch (e: ParseException) {
            e.printStackTrace();
            null
        }
    }

    override fun toData(javaValue: Date): String? {
        // we don't need to implement for the moment
        return null
    }

}